app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java

```diff
@@ -88,4 +141,100 @@ private AppCompatActivity getAppCompatActivity() {
     public Callback getCallback() {
         return FragmentUtil.getCallback(this, Callback.class);
     }
+
+    private void updateUserDestinationList() {
+        if (getCallback() != null) {
+            getCallback().onRequestDestinationListUpdate();
+        }
+    }
+
+    public void setUserDestinationList(List<Trip> trips) {
+        if (destinationAdapter != null) {
+            destinationAdapter.setUserDestination(trips);
+        }
+    }
+
+    public interface Callback {
+        void onPlaceSelected(Place place);
+
+        void onRequestDestinationListUpdate();
+
+        void onDeleteDestination(long list);
+
+        void onDestinationnHistorySelected(int id);
+    }
+
+    // Adapter for the RecyclerView
+    public final class DestinationAdapter extends RecyclerView.Adapter<DestinationItemHolder> {
+        private Context context;
+        private List<Trip> userDestinationList;
+
+        public DestinationAdapter(Context context) {
+            this.context = context;
+            this.userDestinationList = new ArrayList<>();
+        }
+
+        @Override
+        public DestinationItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_travel_destination_picker_saved_destinations, parent, false);
+            return new DestinationItemHolder(v);
+        }
+
+        @Override
+        public void onBindViewHolder(DestinationItemHolder holder, int position) {
+            holder.bindItem(userDestinationList.get(position));
+        }
+
+        @Override
+        public int getItemCount() {
+            return userDestinationList.size();
+        }
+
+        public void setUserDestination(List<Trip> trips) {
+            this.userDestinationList = trips;
+            notifyDataSetChanged();
+            destinationList.getLayoutManager().scrollToPosition(userDestinationList.size() - 1);
+        }
+
+        // Insert a new item to the RecyclerView on a predefined position, could be used in the future
+        public void insert(int position, Trip data) {
+            userDestinationList.add(position, data);
+            notifyItemInserted(position);
+        }
+    }
+
+    public final class DestinationItemHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+
+        @BindView(R.id.destination_view_list)
+        RelativeLayout destinationLayout;
+        @BindView(R.id.destination_name_view_text)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)